sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -448,4 +456,120 @@ export class ApplicationStudentsController extends BaseController {
       applicationNumber: application.applicationNumber,
     };
   }
+
+  /**
+   * Get in progress details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/in-progress")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getInProgressApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<InProgressApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+
+    const incomeVerificationDetails =
+      await this.craIncomeVerificationService.allIncomeVerificationsForAnApplication(
+        applicationId,
+      );
+    const incomeVerification =
+      this.applicationControllerService.processApplicationIncomeVerificationDetails(
+        incomeVerificationDetails,
+      );
+
+    const supportingUserDetails =
+      await this.supportingUserService.getSupportingUsersByApplicationId(
+        applicationId,
+      );
+
+    const supportingUser =
+      this.applicationControllerService.processApplicationSupportingUserDetails(
+        supportingUserDetails,
+      );
+
+    return {
+      id: application.id,
+      applicationStatus: application.applicationStatus,
+      pirStatus: application.pirStatus,
+      PIRDeniedReason: getPIRDeniedReason(application),
+      offeringStatus: application.currentAssessment?.offering.offeringStatus,
+      exceptionStatus: application.applicationException?.exceptionStatus,
+      ...incomeVerification,
+      ...supportingUser,
+    };
+  }
+
+  /**
+   * Get cancelled details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/cancelled")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getCancelledApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<CancelledApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+    return {
+      statusUpdatedOn: application.applicationStatusUpdatedOn,
+    };
+  }
+
+  /**
+   * Get details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/details")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getApplicationStatusDetails(
```

Comment:
Do we need a new method to get this info? Can we use the `getByApplicationId`?

Reply:
`getByApplicationId` It is not take care of PIR waiting scenario (when some properties get their value from application form data), 
![image](https://user-images.githubusercontent.com/77353155/192875968-d8e631e6-404a-46b8-bd37-4559a2eb1eca.png)